### PR TITLE
Test Code Generation with Flutter Version

### PR DESCRIPTION
**Pull request by dickermoshe**
Flutter is not always on the latest version of dart.
We wan't to test that the version of dart that flutter is using is works with drift_dev.
I've also changed the `sqlite3` constraints on the `drift_flutter` package to be compatible with older versions of dart.

I've also added testing for Flutter 3.19. 3.22 changed lots of stuff. IMHO I think we should support the 2 latest version of Flutter.

EDIT:
I've also set better lower version constraints for come packages.
All packages should still work if `pub downgrade` is ran.

### CODE REVIEW

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -207,11 +207,16 @@ jobs:
           dart test
   test_flutter:
     runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        flutter: [3.19, 3.22]
```
Comment (simolus3) [inline .github/workflows/main.yml:213]: Add a comment explaining why these, and when they should be updated.

```suggestion
        # We want to support the two latest stable Flutter versions
        flutter: [3.19, 3.22]
```

**Review by simolus3 — COMMENTED** [review]
Thanks! I agree that having these checks should make releases a lot less surprising and catch a bunch of Flutter incompatibilities during development.

Looks like we only need to change `drift_flutter` to have `^3.3.0` in its `environment.sdk` constraints?

Another thing we may want to test is the drift_dev CLI. Can we run `dart run drift_dev schema dump lib/database/database.dart drift_schemas/`, `dart run drift_dev schema generate drift_schemas/ test/generated_migrations/` and `dart run drift_dev schema steps drift_schemas/ lib/database/schema_versions.dart` in the example app before we run `flutter test`? That would serve as a regression test for https://github.com/simolus3/drift/issues/3097.

**Comment by dickermoshe** [comment]
Chore:
Run `dart pub downgrade` and then pin the lower requirements until all the errors go away.

**Comment by dickermoshe** [comment]
Ready for review

**Review by simolus3 — APPROVED** [review]
LGTM, thanks!

**Pull request merged**